libraries/WiFi/src/WiFiGeneric.h

```diff
@@ -196,6 +202,7 @@ class WiFiGenericClass
     static const char * getHostname();
     static bool setHostname(const char * hostname);
     static bool hostname(const String& aHostname) { return setHostname(aHostname.c_str()); }
+    static int hostByName6(const char *aHostname, ip_addr_t& aResult);
```

Comment:
This method should be private and user should need to call only `hostByName`. You can add extra argument to `hostByName` that can force v6 answer, but if not set, there should be a logic to pick which result to give.

Reply:
Will check

Reply:
Actually I agree, this code comes from another source (can't remember which) and we don't use it in Tasmota.

Reply:
Host are more likely to have multiple IPv6 addresses than multiple IPv4, but either is possible, so rather than one hostByName() with one address results, there probably should be a way to get multiple results.

For a function that only returns one result, it should kind of follow the logic of RFC 3484. Usually that logic is used to select both the source and destination address, given a list of both. e.g. if both have a local scope use that.

However in the case of hostByName we only have the destination address (not the source); as it is a DNS lookup, it is probably also a global address, and for safety we should assume that, i.e. return a global address in preference to a local scope one.

Generally then the algoritm would preference IPv6 over IPv4 over deprecated addresses.

Reply:
I have a very rough algorithm `GsmModemCommon::compareIPAddress` at https://github.com/sgryphon/AdvancedGsmClient/blob/main/src/AdvancedGsm/GsmModemCommon.cpp

Note that it was written pre-support for IPv6 and so operates on raw strings. This means scopes like link-local are determined by "fe80:" prefix, rather than actual bytes. But it could be converted relatively easily.

Reply:
To be fair, I don't use this one anymore in Tasmota. I call directly `dns_gethostbyname_addrtype()` with some custom logic to favor IPv4 or IPv6 depending on configuration of options.
I'm ok with any implementation.